### ISSUE TITLE
InstanceCreditSpecification.NotSupported after create an EC2 instance

### ISSUE THREAD

**Issue by marcusrios** [terraform-providers/terraform-provider-aws#8055]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.11.13
+ provider.aws v2.3.0
```

### Affected Resource(s)

* aws_instance

### Terraform Configuration Files

```hcl
provider "aws" {
  region = "sa-east-1"
}

terraform {
  backend "s3" {
    bucket         = "my-state-bucket"
    key            = "teste-ec2-marcus.tfstate"
    region         = "us-east-1"
    dynamodb_table = "terraform_state_locking"
  }
}

resource "aws_instance" "teste_ec2_marcus" {
  ami               = "ami-04a7ebb302d65e89c"
  availability_zone = "sa-east-1b"
  subnet_id         = "subnet-xxxxx"
  instance_type     = "m5.large"
  key_name          = "my-key"

  tags = {
    Name = "teste-ec2-marcus"
  }
}
```

### Debug Output

https://gist.github.com/marcusrios/76f29baa380c53949d1c602f7411409c

### Panic Output

```
Acquiring state lock. This may take a few moments...
aws_instance.teste_ec2_marcus: Refreshing state... (ID: i-08d1f93f0d955e529)
Releasing state lock. This may take a few moments...

Error: Error refreshing state: 1 error(s) occurred:

* aws_instance.teste_ec2_marcus: 1 error(s) occurred:

* aws_instance.teste_ec2_marcus: aws_instance.teste_ec2_marcus: InstanceCreditSpecification.NotSupported: The instance ID i-08d1f93f0d955e529 does not support T2/T3 Unlimited.
	status code: 400, request id: b1135cdf-b119-4f45-800b-faac4d1b3a0e
```

### Expected Behavior

Refresh instance state without errors.

### Actual Behavior

It's looks like the terraform-provider-aws is trying to get information about cpu_credits for instances that doesn't have this atribute (e.g: `m5-large`). In this case the aws api throws an error `InstanceCreditSpecification.NotSupported`. [See](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstanceCreditSpecifications.html#API_DescribeInstanceCreditSpecifications_Errors)

### Steps to Reproduce

```
terraform init && terraform apply
```

### Important Factoids

No.

### References

https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_instance.go#L867

I'm change the error message to `InstanceCreditSpecification.NotSupported` build a new binary and after that works as expected.

Maybe the AWS api change the error message from `UnsupportedOperation ` to `InstanceCreditSpecification.NotSupported`?

**Comment by marcusrios** [comment]
I'm try to reproduce this behaviour in other region (in my case us-east-1) and I can't. For some reason works fine 🤔 

In us-east-1:
```
➜  teste-ec2-marcus git:(master) ✗ aws ec2 describe-instances --region us-east-1 --instance-ids i-07750d8d0f97a89a5 | jq '.Reservations[].Instances[].InstanceType'
"m5.large"
➜  teste-ec2-marcus git:(master) ✗ aws ec2 describe-instance-credit-specifications --region us-east-1 --instance-id i-07750d8d0f97a89a5
{
    "InstanceCreditSpecifications": [
        {
            "InstanceId": "i-07750d8d0f97a89a5",
            "CpuCredits": "standard"
        }
    ]
}
```

In sa-east-1:
```
➜  teste-ec2-marcus git:(master) ✗ aws ec2 describe-instances --region sa-east-1 --instance-ids i-00d15d464e3106224 | jq '.Reservations[].Instances[].InstanceType'
"m5.large"
➜  teste-ec2-marcus git:(master) ✗ aws ec2 describe-instance-credit-specifications --region sa-east-1 --instance-id i-00d15d464e3106224

An error occurred (InstanceCreditSpecification.NotSupported) when calling the DescribeInstanceCreditSpecifications operation: The instance ID i-00d15d464e3106224 does not support T2/T3 Unlimited.
```

**Comment by ernestm** [comment]
This is currently only affecting sa-east-1, our AWS TAM tells us it’s a new API change that has rolled out to that region first but is coming to others.  It’s breaking our TF operations in that region as well.

**Comment by ernestm** [comment]
We are having this issue in sa-east-1 only as well (out of 10 regions we use).

**Comment by richardboydii** [comment]
I just tested a deployment in sa-east-1 and it looks like the error is no longer occurring. @marcusrios could you verify that you're no longer receiving an error?

**Comment by paultyng** [comment]
My understanding is that this has been rolled back.

**Comment by marcusrios** [comment]
@richardboydii I tested again in sa-east-1 region and now it's working as expected. Thanks all!

**Comment by bflad** [comment]
It turns out we may still need to fix this behavior to prevent future problems. Pull request submitted to disable the call unless the instance type is T2/T3: https://github.com/terraform-providers/terraform-provider-aws/pull/8107

**Comment by bflad** [comment]
The `DescribeInstanceCreditSpecifications` API call will be removed from the `aws_instance` data source and resource for non T2/T3 instance families in version 2.4.0 of the Terraform AWS Provider, releasing later today.

**Comment by bflad** [comment]
The above change has been released in [version 2.4.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#240-march-29-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
